components/style/stylesheets.rs

```diff
@@ -95,6 +97,21 @@ impl CssRule {
     }
 }
 
+impl ToCss for CssRule {
+    fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
+        match *self {
+            CssRule::Namespace(ref lock) => try!(lock.read().to_css(dest)),
+            // TODO: Uncomment here when Manishearth's CSSOM PR(#14190) lands
```

Comment:
Just base it on top of the CSSOM PR for now, and update the css_text getters to use this (there already is commented code to use this)